iceoryx_examples/icediscovery/iox_wait_for_service.cpp

```diff
@@ -0,0 +1,121 @@
+iox::capro::IdString_t event{"Image"};
+
+void printSearchResult(const iox::runtime::ServiceContainer& result)
+{
+    std::cout << "Search result: " << (result.empty() ? "empty" : "") << std::endl;
+
+    for (auto entry : result)
+    {
+        std::cout << "<" << entry.getServiceIDString() << ", " << entry.getInstanceIDString() << ", "
+                  << entry.getEventIDString() << ">" << std::endl;
+    }
+}
+
+Discovery* discoveryPtr{nullptr};
+
+void sigHandler(int)
+{
+    std::cout << "sig handler" << std::endl;
+    if (discoveryPtr)
+    {
+        discoveryPtr->unblockWait();
+    }
+}
+
+int main()
+{
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    //! [create custom discovery]
+    // requires the runtime to be created first
+    Discovery discovery;
+    //! [create custom discovery]
+    discoveryPtr = &discovery;
+
+
+    auto sigTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+    auto sigIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+
+    //! [define search query]
+    auto query = [&]() {
+        auto result = discovery.findService(service, instance, event);
+        return result.size() > 0;
+    };
+    //! [define search query]
+
+    while (!iox::posix::hasTerminationRequested())
+    {
+        std::cout << "Waiting for service <" << service << ", " << instance << ", " << event << "> ..." << std::endl;
+
+        //! [wait until query returns true]
+        bool serviceWasAvailable = discovery.waitUntil(query);
+        //! [wait until query returns true]
+
+        // did we wake up due to an unblock or because the service was available?
+        if (serviceWasAvailable)
+        {
+            std::cout << "<" << service << ", " << instance << ", " << event << "> was available" << std::endl;
+
+            // service was available, but we can never be sure the service is still available
+            // if this is important we need to monitor it (see discovery monitor example)
+
+            std::cout << "Waiting for any discovery change ..." << std::endl;
+
+            do
+            {
+                //! [wait until discovery changes]
+                discovery.waitUntilChange();
+                //! [wait until discovery changes]
+                std::cout << "Discovery changed. Searching <" << service << ", " << instance << ", " << event << "> ..."
+                          << std::endl;
+
+                //! [check service availability]
+                auto result = discovery.findService(service, instance, event);
+                //! [check service availability]
+
+                // we may still have found the service
+                if (result.size() == 0)
+                {
+                    // service was unavailable
+                    break;
+                }
+            } while (true);
+
+            std::cout << "<" << service << ", " << instance << ", " << event << "> was unavailable" << std::endl;
```

Comment:
The `APP_NAME` should be added as prefix to make it testable in the integrationtest. This applies also for the other terminal outputs in this application and the discovery_monitor